### PR TITLE
make passing revoke date in application level revoke token mandatory

### PR DESCRIPTION
**Pull request by SiddhantAgarwal**
## CLA

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [ ] Code changes are tested

## Changelog

- Make passing revoke date for application level revoke mandatory.

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -660,6 +656,12 @@ export class StreamChat<
    * Revokes all tokens on application level issued before given time
    */
   async revokeTokens(before?: Date | string | null) {
+    if (before === undefined) {
+      throw new Error(
+        'Passing before parameter is mandatory for application level token revocation',
+      );
+    }
```
Comment (AnatolyRugalev) [inline src/client.ts:663]: Let's just remove `?` from function arguments

Comment (SiddhantAgarwal) [reply src/client.ts:663]: we can remove `?` but we use the call as plain javascript call for ex - in chat-qa which causes the before param to be undefined.
I think we should have the check for safety.

Comment (AnatolyRugalev) [reply src/client.ts:663]: Well, even in JS linter will provide you with error message. And there's nothing bad in sending undefined anyway as the backend will respond with an error. Having `?` in parameters is misleading as one might think that it is possible to call it without an argument

**Review by AnatolyRugalev — APPROVED** [review]

**Review by ferhatelmas — APPROVED** [review]
LGTM

**Pull request merged**